java/src/org/futo/inputmethod/event/CombinerChain.java

```diff
@@ -58,6 +59,7 @@ public CombinerChain(final String initialText) {
         mCombiners = new ArrayList<>();
         // The dead key combiner is always active, and always first
         mCombiners.add(new DeadKeyCombiner());
+        mCombiners.add(new KoreanCombiner());
```

Comment:
Instead of adding it by default for all layouts, you can now add it as an enum here
https://github.com/futo-org/android-keyboard/blob/f106d515b3eeb426536fb3c4ec1b3f8e11796f72/java/src/org/futo/inputmethod/v2keyboard/CombinerKind.kt#L7

And specify in the layout that you want to use that combiner by the enum name you defined
https://github.com/futo-org/android-keyboard/blob/f106d515b3eeb426536fb3c4ec1b3f8e11796f72/java/src/org/futo/inputmethod/v2keyboard/Keyboard.kt#L265

You can specify different combiners for different layouts now too, if they need to have different combining rules

Reply:
Thank you so much for this. This changes everything

Reply:
Done